JustSaying/AwsTools/MessageHandling/MessageDispatcher.cs

```diff
@@ -47,6 +48,23 @@ public async Task DispatchMessage(SQSMessage message)
             try
             {
                 typedMessage = _serialisationRegister.DeserializeMessage(message.Body);
+
+                typedMessage.MessageAttributes = message.MessageAttributes?.ToDictionary(
+                    source => source.Key,
+                    source =>
+                    {
+                        if (source.Value == null)
+                        {
+                            return null;
+                        }
+
+                        return new JustSaying.Models.MessageAttributeValue
+                        {
+                            StringValue = source.Value.StringValue,
+                            BinaryValue = source.Value.BinaryValue?.ToArray(),
+                            DataType = source.Value.DataType
+                        };
+                    });
```

Comment:
```suggestion
                typedMessage = _serialisationRegister.DeserializeMessage(message.Body);

                Models.MessageAttributeValue MapAttributes(MessageAttributeValue sourceAttribute) =>
                    new Models.MessageAttributeValue
                    {
                        StringValue = sourceAttribute.StringValue,
                        BinaryValue = sourceAttribute.BinaryValue?.ToArray(),
                        DataType = sourceAttribute.DataType
                    };
                
                typedMessage.MessageAttributes = message.MessageAttributes?.ToDictionary(
                    source => source.Key,
                    source => source.Value == null ? null : MapAttributes(source.Value));
```

Reply:
I clearly haven't got the hang of this suggestion thing

Reply:
I smell F#, thx for the snippet 👍